### PR TITLE
[FLINK-16597] handle exception thrown from close method in OutputFormat

### PR DESCRIPTION
**Pull request by Jiayi-Liao**
## What is the purpose of the change

The exception from `format.close()` in finally block in #DataSinkTask#invoke is not properly handled but is only logged. We should make sure the lifecycle of the format inside a try-catch code block and handle the error if anything goes wrong.

## Brief change log

Remove the `format.close()` in finally block and add a new try-catch-finally block inside the original try-catch block.

## Verifying this change

Unit tests to verify the `tryCleanupOnError` is executed when exception is thrown from `format.close()`.

## Does this pull request potentially affect one of the following parts:

It may affect the order betwwen #format.close() and #format.tryCleanupOnError, but this should be expected.

## Documentation

No need.

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/operators/DataSinkTaskTest.java [hunk 1]
```diff
@@ -276,6 +277,25 @@ public void testSortingDataSinkTask() {
 		}
 	}
 
+	@Test
+	public void testClosingErrorDataSinkTask() throws Exception {
```
Comment (rkhachatryan) [inline flink-runtime/src/test/java/org/apache/flink/runtime/operators/DataSinkTaskTest.java:281]: Though the test does check that `close` was called, it doesn't test if the original issues were resolved (order of `close` and `tryCleanupOnError` on `cancel` and `invoke`).

**Review by rkhachatryan — CHANGES_REQUESTED** [review]
Thanks for the contribution @Jiayi-Liao and sorry for the delay with reviewing.

Can you please:
- add Jira ticket id and component to the commit message (see [guidelines](https://flink.apache.org/contributing/code-style-and-quality-pull-requests.html#commit-naming-conventions))
- rebase to master
- update `cancel` method and adjust test (see the comments below)

**Comment by Jiayi-Liao** [comment]
@rkhachatryan I notice that the community is doing the unification of streaming and batch. But this #DataSinkTask can only be used in DataSet job, I'm not sure that this work should be continued.

**Comment by rkhachatryan** [comment]
I think you're right @Jiayi-Liao, the unification lowers the priority of this issue.
But I doubt that this class will be removed anytime soon, nor should there be any conflicts.

@aljoscha, what do you think?

**Comment by aljoscha** [comment]
I think there won't be conflicts, but I'm also not eager to review changes for code that will be removed eventually.

**Comment by rkhachatryan** [comment]
Thanks @aljoscha!
I'm going to close the ticket then. @Jiayi-Liao please reopen it if you'd like this PR to be merged.

**Pull request closed without merging**